### PR TITLE
Option to record with custom freq

### PR DESCRIPTION
**Pull request by gajena**
This provides all the features of [original rosbag](http://wiki.ros.org/rosbag/Commandline) with custom freq recording of topics.

## dependency
* Yaml-CPP :- https://github.com/jbeder/yaml-cpp

## how to record custom rosbag
```shell
rosbag record --file-name record.yaml
```
> this only records and throttles down the topics mentioned in record.yaml
## param 
* --file-name: file which has topics for custom recording

## example:- record.yaml
```yaml
# topic_name: freq(double)
# topic_name: -1 (it will be recorded at publish freq)
#           : 0 (it will not be recorded)
# For topic name same regular expressions cli can be used: ex. "/realsense(.*)" : 0
# [used by default rosbag](http://wiki.ros.org/rosbag/Commandline)
# if there is a topic in a namespace which has been excluded using regular expressions: 
#   to record that topic put that topic before the regular expression
"/chatter": 2
"/raw_odom": 1
"/realsense_bottom/accel/sample": 1
"/realsense_bottom/depth/camera_info": 1
"/realsense_bottom/depth/color/points": 1
"/realsense_bottom/depth/image_rect_raw": 1
"/realsense_bottom/infra1/camera_info": 1
"/realsense_bottom/infra1/image_rect_raw": 1
```

### CODE REVIEW

File: tools/rosbag/CMakeLists.txt [hunk 1]
```diff
@@ -8,19 +8,20 @@ endif()
 find_package(catkin REQUIRED COMPONENTS rosbag_storage rosconsole roscpp std_srvs topic_tools xmlrpcpp)
 find_package(Boost REQUIRED COMPONENTS date_time regex program_options filesystem)
 find_package(BZip2 REQUIRED)
+find_package(yaml-cpp REQUIRED)
```
Comment (dirk-thomas) [inline tools/rosbag/CMakeLists.txt:11]: A proper dependency needs to be declared in the `package.xml` in order to correctly build this package and pass CI.

File: tools/rosbag/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# rosbag
+This is a custom rosbag package which provides all the feature of [original rosbag](http://wiki.ros.org/rosbag/Commandline) with custom freq recording of topics.
```
Comment (dirk-thomas) [inline tools/rosbag/README.md:2]: This doesn't sounds right. This is the readme in the `rosbag` directory - its not a "custom rosbag package".

**Review by dirk-thomas — COMMENTED** [review]
Please consider to add test coverage for the new feature.

**Comment by dirk-thomas** [comment]
@gajena Friendly ping.

**Comment by gajena** [comment]
> @gajena Friendly ping.

Hey @dirk-thomas, sorry for the delay. I will try to get this merged in by the next 3-4 days.

**Comment by dirk-thomas** [comment]
@gajena Just to be aware: the scheduled release of Noetic is May 23rd and we will stop accepting changes a few days before that. Also ABI breaking changes (like this one) won't be accepted into an already released distro and there is no plan for another ROS 1 distro after Noetic.

**Comment by dirk-thomas** [comment]
I will go ahead and close this PR since the patch still has conflicts. And since Noetic has been released in the meantime only API/ABI compatible patches will be considered. Please consider to open a new API/ABI compatible PR instead.

**Pull request closed without merging**